---
title: "Results"
---


```{r}
# Loading all the libraries
library(tidyverse)
library(knitr)
library(ggtree)
library(TDbook) #A Companion Package for the Book "Data Integration, Manipulation and Visualization of Phylogenetic Trees" by Guangchuang Yu (2022, ISBN:9781032233574).
library(ggimage)
library(rphylopic)
library(treeio)
library(tidytree)
library(ape)
library(TreeTools)
library(phytools)
library(ggnewscale)
library(ggtreeExtra)
library(ggstar)
library(DT)
library(plotly)
```

```{r}
NEON_MAGs <- read_csv("data/NEON/GOLD_Study_ID_Gs0161344_NEON_2024_4_21.csv") %>% 
  # remove columns that are not needed for data analysis
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`, `Bin Lineage`)) %>% 
  # create a new column with the Assembly Type
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`,
                            TRUE ~ "Individual")) %>% 
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "d__", "") %>%  
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "p__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "c__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "o__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "f__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "g__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "s__", "") %>%
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), ";", remove = FALSE) %>% 
  mutate_at("Domain", na_if,"") %>% 
  mutate_at("Phylum", na_if,"") %>% 
  mutate_at("Class", na_if,"") %>% 
  mutate_at("Order", na_if,"") %>% 
  mutate_at("Family", na_if,"") %>% 
  mutate_at("Genus", na_if,"") %>% 
  mutate_at("Species", na_if,"") %>% 
  
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "S-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-") 
```

```{r}
NEON_metagenomes <- read_tsv("data/NEON/exported_img_data_Gs0161344_NEON.tsv") %>% 
  select(-c(`Domain`, `Sequencing Status`, `Sequencing Center`)) %>% 
  rename(`Genome Name` = `Genome Name / Sample Name`) %>% 
  filter(str_detect(`Genome Name`, 're-annotation', negate = T)) %>% 
  filter(str_detect(`Genome Name`, 'WREF plot', negate = T)) 
```

```{r}
NEON_metagenomes <- NEON_metagenomes %>% 
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-") 
```

```{r}
NEON_chemistry <- read_tsv("data/NEON/neon_plot_soilChem1_metadata.tsv") %>% 
  # remove -COMP from genomicsSampleID
  mutate_at("genomicsSampleID", str_replace, "-COMP", "") 
```


```{r}
NEON_MAGs_metagenomes_chemistry <- NEON_MAGs %>% 
  left_join(NEON_metagenomes, by = "Sample Name") %>% 
  left_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID")) %>% 
  rename("label" = "Bin ID")
```

```{r}
tree_arc <- read.tree("data/NEON/gtdbtk.ar53.decorated.tree")
tree_bac <- read.tree("data/NEON/gtdbtk.bac120.decorated.tree")
```

```{r}
# Make a vector with the internal node labels
node_vector_arc = c(tree_arc$tip.label,tree_arc$node.label)

# Search for your Phylum or Class to get the node
grep("p__", node_vector_arc, value = TRUE)
```
```{r}
NEON_MAGs_Ind <- read_csv("data/NEON/GOLD_Study_ID_Gs0161344_NEON.csv") %>%
  filter(`Genome Name` != "NEON combined assembly") 
```
```{r}
NEON_MAGs <- read_csv("data/NEON/GOLD_Study_ID_Gs0161344_NEON_edArchaea.csv") %>% 
  # remove columns that are not needed for data analysis
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`)) %>% 
  # create a new column with the Assembly Type
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`,
                            TRUE ~ "Individual")) %>% 
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>% 
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), "; ", remove = FALSE) %>% 
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "S-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-")
```
```{r}

NEON_MAGs_SRER<- NEON_MAGs %>%
  filter(`Site ID`=="SRER")
```

```{r}

NEON_MAGs_Archaea <-NEON_MAGs %>%
  filter(Phylum=="Thermoplasmota")
```


```{r}
NEON_metagenomes <- read_tsv("data/NEON/exported_img_data_Gs0161344_NEON.tsv") %>% 
  rename(`Genome Name` = `Genome Name / Sample Name`) %>% 
  filter(str_detect(`Genome Name`, 're-annotation', negate = T)) %>% 
  filter(str_detect(`Genome Name`, 'WREF plot', negate = T)) 
```

```{r}
NEON_metagenomes <- NEON_metagenomes %>% 
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-") 
```

```{r}
NEON_chemistry <- read_tsv("data/NEON/neon_plot_soilChem1_metadata.tsv") %>% 
  # remove -COMP from genomicsSampleID
  mutate_at("genomicsSampleID", str_replace, "-COMP", "") 
```

```{r}
NEON_MAGs_metagenomes_chemistry <- NEON_MAGs %>% 
  left_join(NEON_metagenomes, by = "Sample Name") %>% 
  left_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID")) 
```

```{r}
NEON_MAGs_metagenomes_chemistry_archaea <- NEON_MAGs_metagenomes_chemistry %>% 
  filter(`Domain.x` == "Archaea")
```
```{r}
# For unknown reasons the following does not like blank spaces in the names
NEON_MAGs_metagenomes_chemistry_noblank <- NEON_MAGs_metagenomes_chemistry %>% 
  rename("AssemblyType" = "Assembly Type") %>% 
  rename("BinCompleteness" = "Bin Completeness") %>% 
  rename("BinContamination" = "Bin Contamination") %>% 
  rename("TotalNumberofBases" = "Total Number of Bases") %>% 
  rename("EcosystemSubtype" = "Ecosystem Subtype")
```

```{r}
NEON_MAGs_metagenomes_chemistry <- NEON_MAGs %>% 
  left_join(NEON_metagenomes, by = "Sample Name") %>% 
  left_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID")) %>% 
  rename("label" = "Bin ID")
```


```{r}
NEON1 <- NEON_MAGs %>%
  full_join(NEON_metagenomes, by = "Sample Name")
NEON_full <- NEON1 %>%
  full_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID"))
```

```{r}
NEON_arc <- NEON_full %>%
  filter(Domain.x =='Archaea')

```

# Results

# Figure 1.

## Phylogenetic Tree: Classes of Archaea in Database

```{r}
ggtree(tree_arc, layout="circular", branch.length="none") +
  
    geom_hilight(node=46, fill="steelblue", alpha=.6) +
    geom_cladelab(node=46, label="Halobacteriota", align=TRUE, hjust=-0.1, offset = 0, textcolor='steelblue', barcolor='steelblue') +

    geom_hilight(node=50, fill="darkgreen", alpha=.6) +
    geom_cladelab(node=50, label="Thermoplasmatota", align=TRUE, vjust=-0.4, offset = 0, textcolor='darkgreen', barcolor='darkgreen') +
  
      geom_hilight(node=54, fill="darkorange", alpha=.9) +
    geom_cladelab(node=54, label="Methanobacteriota", align=TRUE, hjust=1.8, offset = 0, textcolor='darkorange', barcolor='darkorange') +
  
    geom_hilight(node=55, fill="purple", alpha=.6) +
    geom_cladelab(node=55, label="Thermoproteota ", align=TRUE, hjust= 1.0, offset = 0, textcolor='purple', barcolor='purple')
```

# Figure 2.

## Archaea Ecoystem Subtype and Total Number of Bases

```{r}
ggtree(tree_arc, layout="rectangular")  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_point2(mapping=aes(color=`Ecosystem Subtype`, size=`Total Number of Bases`))

```

# Figure 3.

## Ecoystem Subtype and Corresponding Bin Completeness and Contamination

```{r}
# For unknown reasons the following does not like blank spaces in the names
NEON_MAGs_metagenomes_chemistry_noblank <- NEON_MAGs_metagenomes_chemistry %>% 
  rename("AssemblyType" = "Assembly Type") %>% 
  rename("BinCompleteness" = "Bin Completeness") %>% 
  rename("BinContamination" = "Bin Contamination") %>% 
  rename("TotalNumberofBases" = "Total Number of Bases") %>% 
  rename("EcosystemSubtype" = "Ecosystem Subtype")

ggtree(tree_arc)  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_tippoint(aes(colour=`Ecosystem Subtype`)) + 

# For unknown reasons the following does not like blank spaces in the names
  geom_facet(panel = "Bin Completeness", data = NEON_MAGs_metagenomes_chemistry_noblank, geom = geom_point, 
      mapping=aes(x = BinCompleteness)) +
  geom_facet(panel = "Bin Contamination", data = NEON_MAGs_metagenomes_chemistry_noblank, geom = geom_col, 
                aes(x = BinContamination), orientation = 'y', width = .6) +
  theme_tree2(legend.position=c(.1, .7))

```

# Figure 4.

## Phylogenetic Tree Displaying Class

```{r}
ggtree(tree_arc, layout="slanted")  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  
  geom_tiplab(size=2, hjust=-.5) +
  xlim(0,1.5) +
  geom_point(mapping=aes(color=Class)) 
```

# Figure 5.

## Ecosystem Subtype and Corresponding Sample ID

```{r}
ggtree(tree_arc)  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_tiplab(size=2, hjust=-.1) +
  xlim(0,20) +
  geom_point(mapping=aes(color=`Ecosystem Subtype`)) 
```

# Figure 6.

```{r}
ggtree(tree_arc, layout="circular")  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_point2(mapping=aes(color=`Ecosystem Subtype`, size=`Total Number of Bases`))
```

# Figure 7.

## Phylogenetic Tree Displaying Ecosystem Subtypes and Corresponding Assembly Type and Total Number of Bases for Archaea

```{r}
ggtree(tree_arc, layout="circular", branch.length="none") %<+% 
  NEON_MAGs_metagenomes_chemistry + 
  geom_point2(mapping=aes(color=`Ecosystem Subtype`, size=`Total Number of Bases`)) + 
  new_scale_fill() + 
  geom_fruit(
      data=NEON_MAGs_metagenomes_chemistry_noblank,
      geom=geom_tile,
      mapping=aes(y=label, x=1, fill= AssemblyType),
      offset=0.08,   # The distance between external layers, default is 0.03 times of x range of tree.
      pwidth=0.25 # width of the external layer, default is 0.2 times of x range of tree.
      ) 
```

# Figure 8.

## Phylogenetic Tree Displaying Ecosystem Subtypes and Corresponding Assembly Type and Total Number of Bases For Archaea w/ Bars

```{r}
ggtree(tree_arc, layout="circular", branch.length="none") %<+% 
  NEON_MAGs_metagenomes_chemistry + 
  geom_point2(mapping=aes(color=`Ecosystem Subtype`, size=`Total Number of Bases`)) + 
  new_scale_fill() + 
  geom_fruit(
      data=NEON_MAGs_metagenomes_chemistry_noblank,
      geom=geom_tile,
      mapping=aes(y=label, x=1, fill= AssemblyType),
      offset=0.08,   # The distance between external layers, default is 0.03 times of x range of tree.
      pwidth=0.25 # width of the external layer, default is 0.2 times of x range of tree.
      ) + 
  new_scale_fill() +
  geom_fruit(
          data=NEON_MAGs_metagenomes_chemistry_noblank,
          geom=geom_col,
          mapping=aes(y=label, x=TotalNumberofBases),  
          pwidth=0.4,
          axis.params=list(
                          axis="x", # add axis text of the layer.
                          text.angle=-45, # the text size of axis.
                          hjust=0  # adjust the horizontal position of text of axis.
                      ),
          grid.params=list() # add the grid line of the external bar plot.
      ) + 
      theme(#legend.position=c(0.96, 0.5), # the position of legend.
          legend.background=element_rect(fill=NA), # the background of legend.
          legend.title=element_text(size=7), # the title size of legend.
          legend.text=element_text(size=6), # the text size of legend.
          legend.spacing.y = unit(0.02, "cm")  # the distance of legends (y orientation).
      ) 
```

#Figure 9. 
## Bin Quality vs. Bin Completeness in NEO MAGs Database

```{r}
NEON_MAGs %>%
  ggplot(aes(x = `Bin Quality`, y = `Bin Completeness`, color = 'Red')) +
  geom_point() +
  labs(x = "Bin Quality", y = "Bin Completeness", title = "Bin Quality Vs. Bin Completeness")
```

#Figure 10. 
## Total Number of Bases vs. Bin Completeness

```{r}
NEON_MAGs %>%
  ggplot(aes(x = `Total Number of Bases`, y = `Bin Completeness`)) +
  geom_point() +
  labs(x = "Total Number of Bases", y = "Bin Completeness", title = "Total Number of Bases Vs. Bin Completeness")

```

#Figure 11. 
## Bin Completeness and Total Number of Bases at Each Tippoint

```{r}
tree_arc <- read.tree("data/NEON/gtdbtk.ar53.decorated.tree")
ggtree(tree_arc, layout="circular")  %<+%
  NEON_MAGs + 
  geom_point2(mapping=aes(color=`Bin Completeness`, size=`Total Number of Bases`))

```

# Figure 12.

## Sample IDs Node 55

```{r}
NEON_metagenomes <- read_tsv("data/NEON/exported_img_data_Gs0161344_NEON.tsv") %>% 
  select(-c(`Domain`, `Sequencing Status`, `Sequencing Center`)) %>% 
  rename(`Genome Name` = `Genome Name / Sample Name`) %>% 
  filter(str_detect(`Genome Name`, 're-annotation', negate = T)) %>% 
  filter(str_detect(`Genome Name`, 'WREF plot', negate = T))
tree_arc_preorder <- Preorder(tree_arc)
tree_Archaea <- Subtree(tree_arc_preorder, 55)
ggtree(tree_Archaea) +
  geom_tiplab(size=2) +
  xlim(0,20)

```

#Figure 13. 
## Archaea Phylogenetic Tree and Sample ID

```{r}
ggtree(tree_Archaea, layout="circular") + 
  geom_tiplab(aes(angle=angle))+
    theme_tree() +
    xlim(0,20)
```

#Figure 14. 
## Archaea Tree and Sample ID

```{r}
ggtree(tree_Archaea, layout="circular")  %<+%
  NEON_MAGs + 
  geom_tiplab(size=2, hjust=-.1) +
  xlim(0,20) 
```

#Figure 15. 
## Bin Completeness at Each Tip Point

```{r}
ggtree(tree_Archaea)  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_tippoint(aes(colour=`Bin Completeness`)) + 
  scale_colour_gradient(low='blue', high='red') +
  geom_tiplab(size=1, hjust=-.1) +
  xlim(0,15)
```

#Figure 16. 
## Bin Completeness and Total Number of Bases at Tip point

```{r}
ggtree(tree_Archaea, layout="circular")  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_point2(mapping=aes(color=`Bin Completeness`, size=`Total Number of Bases`))
```

#Figure 17.

## Phlyum Counts at Santa Rita Experimental Range

```{r}
NEON_MAGs_SRER %>%
  ggplot(aes(y=Phylum))+
  geom_bar()+
  labs(title = "Phylum Counts at Santa Rita Experimental Range")
```

# Figure 18.

## Phylum Size at Santa Rita Experimental Range

```{r}
NEON_MAGs_SRER %>%   
ggplot(aes(x = `Total Number of Bases`, y = `Phylum`)) +
  geom_point()+
  labs(title = "Phylum Size at Santa Rita Experimental Range")
```

# Figure 19.
# Phylum Counts at Santa Rita Experimental Range 

```{r}
ggplotly(
  ggplot(data = NEON_MAGs_SRER, aes(x = `Gene Count`, y = `Scaffold Count`)) +
    geom_point(aes(color = Phylum, shape = Phylum))+
    labs(title = "Phylum Counts at Santa Rita Experimental Range")
 )
```

#Figure 20.

```{r}
NEON_MAGs_metagenomes_chemistry_archaea %>% 
  ggplot(aes(x=`Site ID.x`, y = soilTemp)) +
    geom_boxplot()
```

# Figure 21.
# Order vs. Ecosystem Subtype and Soil Temp 

```{r}
NEON_MAGs_metagenomes_chemistry_archaea %>% 
  ggplot(aes(x=`Ecosystem Subtype`, y = soilTemp, color = Order)) +
    geom_point() +
   theme(axis.text.x = element_text(angle=45, vjust=1, hjust=1))
```

#Figure 22.
# Organic Percentage vs. Nitrogen Percentage by Phylum 

```{r}
ggplotly(
ggplot(data= NEON_full ,aes(x = organicCPercent, y = nitrogenPercent)) +
geom_point(aes(color= Phylum))
)
```

# Figure 23.

## Soil Water pH vs. SoilInCaClpH

```{r}
ggplotly(
ggplot(data= NEON_arc ,aes(x = soilInWaterpH , y = soilInCaClpH)) +
geom_point(aes(color= Family))
)
```
